src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -158,3 +164,39 @@ def _resolve_multi(interpreter, requirements, platforms, find_links):
       allow_prereleases=python_setup.resolver_allow_prereleases)
 
   return distributions
+
+
+def inject_req_libs_provided_by_setup_file(build_graph, local_built_dists, synthetic_address, binary_tgt=None):
+  """Build a requirements library from a local wheel.
+
+  :param build_graph: The build graph needed for injecting synthetic targets.
+  :param local_built_dists: A list of paths to locally built wheels to package into
+  requirements libraries.
+  :param synthetic_address: A generative address for addressing synthetic targets.
+  :param binary_tgt: An optional parameter to be passed only when called by the `python_binary_create`
+  task. This is needed to ensure that only python_dist targets in a binary target's closure are included
+  in the binary for the case where a user specifies mulitple binary targets in a single invocation of
+  `./pants binary`.
+  :return: a :class: `PythonRequirementLibrary` containing a local wheel and its
+  transitive dependencies.
+  """
+  req_libs = []
+  local_whl_reqs = []
+  for whl_location in local_built_dists:
+    should_create_req = False
+    if binary_tgt:
+      if any([tgt.id in whl_location for tgt in binary_tgt.closure()]):
+        should_create_req = True
+    else:
+      should_create_req = True
+    if should_create_req:
+      base = os.path.basename(whl_location)
+      whl_dir = os.path.dirname(whl_location)
+      whl_metadata = base.split('-')
+      req_name = '=='.join([whl_metadata[0], whl_metadata[1]])
+      local_whl_reqs.append(PythonRequirement(req_name, repository=whl_dir))
+  if local_whl_reqs:
+    addr = Address.parse(synthetic_address)
+    build_graph.inject_synthetic_target(addr, PythonRequirementLibrary, requirements=local_whl_reqs)
+    req_libs = [build_graph.get_target(addr)]
+  return req_libs
```

Comment:
the logic here seems fine but I think this function would be more readable/maintainable if you decomposed it a bit - something along the lines of (note the renamed function name + docstring, too):

```
def inject_synthetic_dist_requirements(build_graph, local_built_dists, synthetic_address, binary_tgt=None):
  """Injects synthetic `PythonRequirement` objects from a `PythonDistribution` into the `BuildGraph`.
  ...
  """
  def should_create_req(bin_tgt, loc):
    if not bin_tgt:
      return True
    # <-- Add comment here about why this is necessary
    return any([tgt.id in loc for tgt in bin_tgt.closure()])

  def python_requirement_from_wheel(path):
    base = os.path.basename(path)
    whl_dir = os.path.dirname(path)
    whl_metadata = base.split('-')
    req_name = '=='.join([whl_metadata[0], whl_metadata[1]])
    return PythonRequirement(req_name, repository=whl_dir)

  local_whl_reqs = [
    python_requirement_from_wheel(whl_location)
    for whl_location in local_built_dists
    if should_create_req(binary_tgt, whl_location)
  ]

  if not local_whl_reqs:
    return []

  addr = Address.parse(synthetic_address)
  build_graph.inject_synthetic_target(addr, PythonRequirementLibrary, requirements=local_whl_reqs)
  return [build_graph.get_target(addr)]
```